[MIG] contract: Migration to 14.0

ozono opened the pull request:
contract migration

funbaker committed:
[IMP] Add type to analytic account / contract

gorozcoh committed:
[10.0][FIX] Correct domain attribute in field journal_id (#120)

* Correct domain attribute in field journal_id

Original domain includes unknown value company_id. Throws error when selecting the journal.

* Corregidos errores detectados por Lint

* Refactoring, DRY

* [FIX] Add missing field company_id to account_analytic_contract

* Small refactoring for company_id field

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[FIX+IMP] contract: Real fix to company_id missing + improvements
* company_id was empty because an onchange, not inheritance nor visibility
* Added multi-company group to company_id fields
* Added multi-company access rule to contract templates
* Fix double %% in XML dates that was causing an error
* When creating a contract, recurring_invoices is set by default

pedrobaeza committed:
[10.0][FIX+IMP] contract: Improve usability and don't fail on wrong data (#130)

* [FIX+IMP] contract: Improve usability and don't fail on wrong data

* Cron create invoices masked for avoiding silent errors
* New constraints for assuring data consistency
* UI helps for entering consistent data
* Spanish translation
* Remove double company_id field on form

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] contract: Automatic prices for lines

pedrobaeza committed:
[FIX] contract: Correct references in the contract report

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[IMP+FIX] contract: Several things:

* [IMP] Add computed dates from/to period invoiced (#140)
* [IMP] Improve partner contract smartbutton
* [FIX] Onchange contract template raise error
* [FIX] Invalid pricelist name

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[IMP] contract: Add extend existing invoice possibility

levkar committed:
[FIX] contract: Let billing users have access rights for contracts (#159)

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[IMP] contract_variable_quantity: Recompute price when changing qty

If you have contract lines with automatic price and your pricelist
contains different prices per quantity, the price is not changed
when computing quantity.

This PR fixes this.

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[FIX] contract: Custom values overwrited by xml, missed noupdate

oca-travis committed:
[UPD] Update contract.pot

katyukha committed:
[FIX] analytic invoice/contract lines inheritance

Bug description
---------------

`account.analytic.contract.line` inherits
`account.analytic.invoice.line`

`account.analytic.invoice.line` defines field `analytic_account_id`:
   - comodel='account.analytic.account'

`account.analytic.contract.line` redefines field `analytic_account_id`:
   - comodel='account.analytic.contract'

On attempt to extend `account.analytic.invoice.line` model adding
field that depends on `analytic_account_id.date_start`
Odoo fails to update, because it adds this field to
`account.analytic.contract.line` through inheritance,
and `account.analytic.contract` model have no this field.

What is done
------------

Change inheritance order:
- `account.analytic.invoice.line` inherits
`account.analytic.contract.line`
- no file renames at this stage (this wil be done in next commit)

MiquelRForgeFlow committed:
[IMP] Make sure it works for sale and purchase contracts

jonerikceberio committed:
[IMP] contract: possible error in migration if the column does not exist

katyukha committed:
Rename account_analytic_*_line files

In previous commit changed inheritance order of
'account.analytic.*.line' models, thus classes and models were renamed.

This commit only renames files to temporary names.

This commit does not change file contents.

oca-travis committed:
[UPD] Update contract.pot

RoelAdriaans committed:
[IMP] Fix README.rst

Runbot URL in old README.rst pointed to 10.0 runbot,
fixed by upgrading template

RoelAdriaans committed:
Translated using Weblate (Dutch)

Currently translated at 96.0% (96 of 100 strings)

Translation: contract-11.0/contract-11.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract/nl/

Commit:
Translated using Weblate (German)

Currently translated at 53.0% (53 of 100 strings)

Translation: contract-11.0/contract-11.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract/de/

oca-travis committed:
[UPD] Update contract.pot

Commit:
Translated using Weblate (German)

Currently translated at 75.0% (75 of 100 strings)

Translation: contract-11.0/contract-11.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract/de/

OCA-git-bot committed:
[UPD] README.rst

etobella committed:
Translated using Weblate (Spanish)

Currently translated at 98.0% (98 of 100 strings)

Translation: contract-11.0/contract-11.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract/es/

dreispt committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 10.0% (10 of 100 strings)

Translation: contract-11.0/contract-11.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-11-0/contract-11-0-contract/pt_PT/

pedrobaeza committed:
[IMP] contract: Don't depend on external data

The test as it was, leaves to the demo pricelist the control on the price of
the product, so other modules that modifies this pricelist will make the
test to fail.

This is the minimum change needed for avoiding the problem.

Commit:
[FIX] Contract Spanish Translation

    - Module: contract
    - Summary: several terms where not translated to Spanish.

sbejaoui committed:
[MIG] contract: Migration to 12.0

Commit:
Translated using Weblate (German)

Currently translated at 99.0% (99 of 100 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/de/

sbejaoui committed:
[IMP] - Make recurrence mechanism on contract line

Make recurrence mechanism on contract line and some other refactoring

[FIX] - Keep contract_cron on account_analytic_account model

contract_cron defined with no_update option.
Changing it, will cause issue to past version installation.

[IMP] - Fix recurring_next_date default value

recurring_next_date should have start_date as default value in prepaid policy
and start_date + invoicing_interval if postpaid

[FIX] - Fix test check no journal

[IMP] - Return created invoices on recurring_create_invoice

[IMP] - Specific process to compute recurring_next_date for  monthly-last-day

fixes: #198

[ADD] - Add Post-migration script to bring recurrence info from contract to contract lines

[ADD] - Add search filter based on date_end and recurring_next_date

 - not_finished filter in contract search view
 - finished filter in contract search view
 - Next Invoice group by in contract search view

[ADD] - Add unit tests

- cases to compute first recurring next date
- contract recurring_next_date
- contract date_end

[IMP] - Improve Unit tests

sbejaoui committed:
[12.0][IMP] - Add strat/stop wizard to contract line

[12.0][IMP] - Add pause button to contract line

[IMP] - Add state filed in contract line form

[FIX] - stop don't change date_end for finished contract line

[IMP] - Change contract line buttons visibility

Add renewal process with termination notice

[FIX] - don't consider stop_date If it is after the contract line end_date

[IMP] - consider more cases in stop_plan_successor

[IMP] - cancel upcoming line on stop

[IMP] - Chnage next invoice date on un-cancel

[IMP] - Post message in contract on contract line actions

[IMP] - check contract line overlap

sbejaoui committed:
[FIX] - invoice last period for post-paid case

sbejaoui committed:
[IMP] - Add primary views for contract

[IMP] - don't use related filed for partner_id and pricelist_id

sbejaoui committed:
[FIX] - fix stop_plan_successor case 5

contract line start in the suspension period and end after it

sbejaoui committed:
[IMP] - improve cancel/uncancel process

[FIX] - Test if start_date is set before compute

[FIX] - date_end include in the period in auto_renew case

[FIX] - in suspension case, contract line should start a day after the end

[IMP] - confirm message on contract line cancel

sbejaoui committed:
[IMP] - hide recurring_invoicing_type if recurring_rule_type is monthlylastday

for the monthlylastday case, pre-paid is logicly impossible,
if monthlylastday is set, we consider only post-paid case

[IMP] - Improve unit tests

sbejaoui committed:
[IMP] - store last_date_invoiced on contract_line

Improve CRITERIA_ALLOWED_DICT

[IMP] - code improvement

[IMP] - Use last_date_invoiced to set marker in invoice description

[IMP] - add migration script to init last_day_invoiced and some other improvement

[FIX] - a contract line suspended should start a day after the suspension end

sbejaoui committed:
[IMP] - don't allow to unlink uncnaceled contrac line

sbejaoui committed:
[FIX] - check date_start before onchange

sbejaoui committed:
[FIX] - compute recurring_next_date for contract

[IMP] - get contract line default data onchange product_id

sbejaoui committed:
[IMP] - Add responsible to contract form view

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

sbejaoui committed:
[FIX] - contract recurring_next_date ignore canceled lines

[FIX] - fix _get_invoiced_period if recurring_next_date manually updated

sbejaoui committed:
[IMP] - archive contract_line on contract archive

ThomasBinsfeld committed:
[REF] Contract: invoice creation

[FIX] - Fix typo

[IMP] - date start required for contract line

[REF] Gitignore: .eggs

[REF] Contract Unit Tests: base the cron check on invoice lines instead of invoices

sbejaoui committed:
[FIX] - as date_start is required, constraints on it become useless

[FIX] - for finished contract _get_lines_to_invoice should return False

sbejaoui committed:
[FIX] - default value for active field in contract line

[FIX] - fix flake8

ThomasBinsfeld committed:
[ADD] Contract: payment term

sbejaoui committed:
[12.0][IMP] - check invoice line vals before assignment

[FIX] - Fix unit tests.

sbejaoui committed:
[FIX] - do not copy last_date_invoiced on plan_successor

sbejaoui committed:
[FIX] - renew only recurring_invoices contract

sbejaoui committed:
[FIX] - filter by termination_notice for contract line to renew

sbejaoui committed:
[IMP] - Improve inprogress search filter

sbejaoui committed:
[IMP] - Link invoice line to contract lines

[IMP] - Add index on contract line

[IMP] - Add is_suspended flag and _search_state

[IMP] - Add is_suspended flag to stop contract line

sbejaoui committed:
[MV] - move onchange method to contract_product module

[RMV] - Remove useless check

sbejaoui committed:
[IMP] - Replace is_suspended flag by manual_renew_needed

Add a computed field for the first date of the termination notice period

Adapt state compute and search method

[IMP] - Improve unit tests

ThomasBinsfeld committed:
[FIX] Contract: copy payment term on generated invoice

sbejaoui committed:
[IMP] - Should be able to stop a past contract line

ThomasBinsfeld committed:
[ADD] Contract: fiscal position

[ADD] Contract: new contributor

[REF] Contract: indexes

sbejaoui committed:
[FIX] - recurring_next_date is required if date_end is null

sbejaoui committed:
[IMP] - rename recurring_interval to Invoice Every

ThomasBinsfeld committed:
[IMP] Contract: matching invoice view depending on contract type

[ADD] Contract: invoice partner

sbejaoui committed:
[ADD] - Add more value for auto_renew_rule_type

[REF] - refactoring for compute_first_date_end

[RMV] - remove auto_renew_rule_type option monthlylastday

[IMP] - rename manual_renew_needed label

[IMP] - use get_product_multiline_description_sale to get contract line description

JordiBForgeFlow committed:
[FIX] Errors in navigation to customer invoices / vendor bills

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (109 of 109 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es/

mvrodriguez committed:
Translated using Weblate (Galician)

Currently translated at 100.0% (109 of 109 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/gl/

ThomasBinsfeld committed:
[FIX] Contract to Invoice: take contract vendor instead of partner's

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (109 of 109 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/de/

sbejaoui committed:
[FIX] - manual_renew_needed should be visible to all contract users

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/

sbejaoui committed:
[REF] - add _get_next_invoiced_period method

sbejaoui committed:
[FIX] - Fix In progress contract filter

sbejaoui committed:
[REF] - add method to get quantity to invoice

the contract line quantity can be in some use cases variable, to simplify
the way it is changed we add a new method _get_quantity_to_invoice.

sbejaoui committed:
[FIX] - Onchange contract line don't reste contract lines

[FIX] - Rename _get_invoiced_period to _get_period_to_invoice

[ADD] - Add stop_at_date_end to _get_period_to_invoice

[FIX] - Remove useless filter

sbejaoui committed:
[FIX] - don't play onchange date_start for old lines on contract template change

[FIX] - Fix stop post message

[FIX] - Fix sale_contract_count should count all partner contract

[FIX] - set recurring_next_date to False if contract line stoped at last date invoiced

[FIX] - Group by next_invoice also considers dates in the past

sbejaoui committed:
[IMP] - A canceled contract line can't be set to auto-renew

[REF] Contract: pep8

[IMP] Contract: update USAGE section of README

[IMP]call onchange_is_auto_renew when changing template

OCA-git-bot committed:
[ADD] icon.png

AaronHForgeFlow committed:
[IMP]pass the payment term to the contract from the partner

ThomasBinsfeld committed:
[REF] Contract: split from analytic account

ThomasBinsfeld committed:
[REF] Contract: use context_today instead of time in filters

ThomasBinsfeld committed:
[REF] Contract Sale Invoicing: split from analytic account

[REF] Contract Sale Invoicing: update translations

[IMP] - Assert that the predecessor is available for new link at uncancel

[RMV] - remove usless changes

[RMV] - Remove usless field recurring_invoices

after the total isolation between contract model and account analytic one.
recurring_invoices which was used to mark analytic account as contract became usless

[IMP] - P3 syntax

[IMP] - use @openupgrade.migrate() and openupgrade.logged_query

[IMP] - drop transient table in migration script

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 83.5% (91 of 109 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/it/

sbejaoui committed:
[IMP] - Make one migration from 12.0.1 to 12.0.4

[RMV] - Remove empty file

[IMP] - Update no_update cron after migration

[IMP] - move contract template recurrence info to line level

[FIX] - Fix contract line model description

[IMP] - Link contracts to analytic accounts

[FIX] - Fix pylint

[IMP] - Move chatter and attachments from analytic account to contract

[IMP] - Move account_analytic_id to contract line level

[IMP] - Improve version check in migration script

[IMP] - Move contracts followers from analytic accounts

[ADD] - Add mail.activity.mixin to contract.contract model

remove data drop from migration scripts

[12.0][FIX] - Fix _init_last_date_invoiced

fix flake8

[ADD] - Update contributors list

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (109 of 109 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fr/

hparfr committed:
Add commercial_partner_id fields into contract module

Remove string attribute

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (109 of 109 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 51.8% (57 of 110 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt_BR/

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100.0% (110 of 110 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/de/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 40.9% (45 of 110 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (110 of 110 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/zh_CN/

liweijie0812 committed:
Remove invalid Chinese translation files

pedrobaeza committed:
[FIX+IMP+MIG] contract: Several refinements:

* Remove incorrect oldname attributes.
* Add filter on partners for running contracts (+ a support o2m field for that).
* Cover more tables in model renaming + cleaner code using a loop.
* Don't copy contract lines, but rename table + copy contract records on pre.
* Contract code is now populated to "Reference/Description" field in invoice.
* Order on new contract model has been restored to the same as old analytic accounts.

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (211 of 211 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/zh_CN/

pedrobaeza committed:
[MIG+FIX] contract: Missing pieces for moving info from analytic account to contract

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 96.2% (204 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/it/

pedrobaeza committed:
[MIG] contract: Assign old partner's salesman to migrated contracts

As v11 takes salesman from linked partner and now the salesman is a field in the
contract that is initialized to current user, we need to assign to the recently
converted contracts following old logic, or they will have admin as responsible.

marylla committed:
Translated using Weblate (German)

Currently translated at 78.8% (167 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/de/

albertpons committed:
Translated using Weblate (Catalan)

Currently translated at 61.3% (130 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/ca/

enriquemartin committed:
[12.0][FIX] contract & contract_sale - security fix

The lowest model access for contract.contract model is group_account_invoice group. However the addon adds two smart buttons to res.partner view without any security restrictions and without compute_sudo attribute on computed fields.
This causes the view to crash when a user without the proper permissions tries to access the res.partner form view.
The solution adds groups_id to the partner form views in which the buttons are added, so the only loads when the user has proper permissions.
Other way to solve it would be to add compute_sudo attribute to the relevant fields, but this causes an access error when the user clicks on the smart buttons.

britoederr committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (212 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt_BR/

pedrobaeza committed:
[MIG] contract: Missing renaming of invoice link

Done through `oldname`.

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (212 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/zh_CN/

pedrobaeza committed:
[MIG] contract: Create contract sequence for allowing to create records

thomaspot committed:
Translated using Weblate (Dutch)

Currently translated at 98.1% (208 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/nl/

sbejaoui committed:
[IMP] - Add contract tags

sbejaoui committed:
[IMP] - Add multi-company access rule for contract objects

sbejaoui committed:
[12.0][FIX] - contract company must be the same as the sale order

When creating a contract from a sale order the company must be the sale order
company and not the user company

sbejaoui committed:
[12.0][FIX] - Fix contract report

rschnapka committed:
Translated using Weblate (German)

Currently translated at 99.1% (210 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/de/

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (212 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es/

sbejaoui committed:
[12.0][FIX] - Fix performance issue on contract line read

multi-company record rules on contract line causes performance issue

sbejaoui committed:
[12.0][FIX] - Fix contract partner label

the label Partner (always False) don't make any sens

sbejaoui committed:
[12.0][ADD] - Add note field to contract

pedrobaeza committed:
[MIG] contract: No contract on AA without recurring invoices flag

Don't transfer to contract those analytic accounts with the "Recurring invoicing" flag
unchecked.

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 21.7% (46 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/hr/

sbejaoui committed:
[12.0][IMP] - Add an action for contracts manual invoicing

It happen that a company has to trigger the invoicing action to generate invoices before
the scheduled date (to print and prepare invoices documents, check invoices, etc.).
This requires technical access for end users with the risk that this represents.

This commit adds a new wizard to run the invoicing action for a given date with a helper
to see and check the contract that will be invoiced. When the manual action is called,
the system displays all created invoices.

[12.0][IMP] - log the manual invoice action in contract chatter

[IMP] - Add alink to the invoice in contract message at manual invoicing

[IMP] - Improve code

[FIX] - log message for invoice creation only when there is an invoice

[IMP] - split the manual invoice menu into to menus sale & purhcase

[IMP] - hide invoice button if there is nothing to invoice

sergio-teruel committed:
[IMP] contract: Fix test crash with other modules

pedrobaeza committed:
[MIG][FIX] contract: Move contracts, but disable them

There were an error in previous query for moving only contracts with the mark checked,
but it's also more logic to move them, but remain them disabled.

sbejaoui committed:
[12.0][FIX] - Run onchange contract_type when changing the contract template

Fix this use-case:

If the contract journal is not set on the contract template the contract is created
without a journal when confirming the sale order

pedrobaeza committed:
[FIX] contract: Don't execute onchange after invoice creation

Using that approach (that is the current one in core)  has a lot of side effects and
performance bottlenecks. You can read odoo/odoo#40156 for summarizing them.

This also improves the handling of the values of payment term an fiscal position
for using the partner ones if not set.

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 22.2% (47 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/hr/

pedrobaeza committed:
[FIX] contract: Tests with duplicated name

So they are not executed at all. Detected by chance looking for a test for the other PR.

sbidoul committed:
[REF] contract: rename misnamed methods

sbidoul committed:
[REF] contract: clarify _get_recurring_next_date

First compute the next period end date,
then derive the next invoice date from the next
period stard and end date.

Martronic-SA committed:
Translated using Weblate (French)

Currently translated at 40.1% (85 of 212 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fr/

sbidoul committed:
[REF] contract: handle max_date_end in _get_recurring_next_date

This concentrates all next date calculation
logic in one place, and will allow further simplifications.

sbidoul committed:
[IMP] contract: add next period start/end fields

Add two computed field showing the next period
start and end date. This improve the UX and will
enable further simplifications in the code.

sbidoul committed:
[REF] contract: refactor _get_period_to_invoice

Move the part of the logic that compute the next
period depending on the chosen next invoice date
to _get_next_period_date_end.

sbidoul committed:
[REF] contract: refactor _update_recurring_next_date

Reuse the logic that is now fully located in _get_recurring_next_date.

sbidoul committed:
[REF] contract: re-add _compute_first_recurring_next_date

For backward compatibility

sbidoul committed:
[FIX] contract: add missing dependency in computed field

sbidoul committed:
[REF] contract: remove one monthlylastday special case

get_relative_delta now works the same for all recurring rules.
Move the special case handling to _init_last_date_invoiced
which is used only for migration.

sbidoul committed:
[IMP] contract: support pre-paid for monthlylastday

monthlylastday is (almost) not a special case anymore \o/.
montlylastday is simply a montly period where the
periods are aligned on month boundaries.
The last bit of special casing is that postpaid generates
invoice the day after the last dasy of the period, except
for monthlylastday where the invoice is generated on the
last day of the period. This last exception will disappear
when we put the offset under user control.

This is a breaking change because the post-paid/pre-paid
mode becomes relevant for monthlylastday invoicing.
The field becomes visible in the UI. Code that generate
monthlylastday contract lines must now correctly set
the pre-paid/post-paid mode too. Some tests have had
to be adapted to reflect that.

sbidoul committed:
[REF] contract: make recurring_invoicing_offset a computed field

In preparation to making it user modifiable.

sbejaoui committed:
[FIX] contract: Improve unit tests

sbidoul committed:
[REF] contract: make get_next_period_date_end public

Make it public because it is the core logic of the module.
Also, clarify that recurring_invoicing_type
and recurring_invoicing_offset are needed only when
we want the next period to be computed from a
user chosen next invoice date.

sbidoul committed:
[REF] contract: rename _get_recurring_next_date as get_next_invoice_date

It is easier to understand. Also make it public.

sbejaoui committed:
[IMP] contract: add unit test for different combinations for next invoicing period

sbejaoui committed:
[REF] contract: simplify _get_period_to_invoice

sbidoul committed:
[IMP] contract: display invoicing offset

augustodinizl committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (216 of 216 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt_BR/

sbejaoui committed:
[12.0][IMP] - Add failing test for wrong invoicing date after contract line stop

sbejaoui committed:
[12.0][FIX] - stop update recurring_next_date

sbejaoui committed:
[12.0][IMP] - add new option: create_new_line_at_contract_line_renew

Add a company config option to decide whether to create or to extend contract
line at renew action

sbejaoui committed:
[12.0][IMP] - extend contract line at renewal

sbejaoui committed:
[IMP] - improve code: unify methods argument _renew_create_line and _renew_extend_line

sbejaoui committed:
[12.0][IMP] - Add failing test for next invoice date before the last date invoiced

sbejaoui committed:
[12.0][FIX] - raise an error when next invoice date before the last date invoiced

scigghia committed:
Translated using Weblate (Italian)

Currently translated at 91.9% (204 of 222 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/it/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/

pedrobaeza committed:
[IMP] contract: remove redundant method

I have detected a method that was created as redundant and with the same
technique used when preparing the line values, so better to have
everything together in the same method instead of having it spread.

pedrobaeza committed:
[FIX] contract: Handle properly multi-company setups

If you have contracts in several companies, cron will create all of them, but
property fields will be populated with incorrect data as the taken company is
the main from the cron user (usually admin).

gregory-moka committed:
Translated using Weblate (French)

Currently translated at 38.1% (85 of 223 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fr/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/

Commit:
Added translation using Weblate (Spanish (Chile))

Currently translated at 0.9% (2 of 223 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es_CL/

Commit:
Translated using Weblate (Spanish)

Currently translated at 99.6% (222 of 223 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/

qgroulard committed:
Translated using Weblate (French)

Currently translated at 38.6% (86 of 223 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fr/

sbejaoui committed:
[IMP] - contract termination

Commit:
Translated using Weblate (Italian)

Currently translated at 91.0% (213 of 234 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/it/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 96.6% (226 of 234 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt_BR/

sbejaoui committed:
[12.0][IMP] - Improve partner With running contracts search view filter

jarmokortetjarvi committed:
Translated using Weblate (Finnish)

Currently translated at 100.0% (234 of 234 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fi/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Translated using Weblate (Portuguese)

Currently translated at 99.6% (233 of 234 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/

jarmokortetjarvi committed:
Translated using Weblate (Finnish)

Currently translated at 100.0% (234 of 234 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fi/

Commit:
[IMP] contract: add analytic_tag_ids field in contract lines

svalaeys committed:
Translated using Weblate (French)

Currently translated at 100.0% (254 of 254 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/fr/

Commit:
[IMP] contract: add section and notes in contract line

svalaeys committed:
Set company_id earlier in the invoice preparation process so that _onchange_partner can execute as intended.

Commit:
Translated using Weblate (Spanish)

Currently translated at 94.9% (241 of 254 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es/

pedrobaeza committed:
[FIX] contract: Delete cascade line wizard

For avoiding problems while trying to remove the contract/contract line

sbejaoui committed:
[IMP] - Place id attribute before model

Co-Authored-By: Carlos Dauden <carlos.dauden@tecnativa.com>

sbejaoui committed:
[IMP] - Contract: add new recurring_rule_type quarterly and semesterly

pedrobaeza committed:
[FIX+IMP] contract: improve test coverage with new sections + fix search method

Commit:
[FIX] contract: Select the correct sections/notes to be invoiced

marylla committed:
Translated using Weblate (German)

Currently translated at 91.3% (240 of 263 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/de/

JorgeQuinteros committed:
Translated using Weblate (Spanish)

Currently translated at 89.3% (241 of 270 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es/

pedrobaeza committed:
[IMP] contract: Remove roadmap

It's already attended.

pedrobaeza committed:
[FIX] contract: Currency is not editable + pricelist from partner

Previous related field was not accurated nor editable. Now the field is got
properly from a computed field.

Reviewing this, as the currency was taken (and it continues being taken) from the partner
pricelist if no pricelist is explicitly set, automatic price should use the same logic for
using partner pricelist.

pedrobaeza committed:
[IMP] contract: Allow to set a manual currency

Done through an extra field + inverse in the existing compute. This way, we have
the flexibility of auto-remove custom value if it matches the computed one + we don't
need extra changes in the rest of the code.

fcolus committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (270 of 270 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/pt_BR/

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (270 of 270 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/es/

Commit:
[FIX] contract: do not die if modified partner has contracts in several companies

In a multicompany scenario where a contact belongs to a company and has contracts in several companies, if the user goes to the contact to edit anything, when saving, it will trigger the recomputation of the `commercial_partner_id` field, which gets populated to all the related contracts, and will undeniably fail with an `AccessError`.

A simple test is provided that, without the fix, fails like this:

```
2020-06-25 07:21:24,859 33 ERROR devel odoo.addons.contract.tests.test_contract: ERROR: test_multicompany_partner_edited (odoo.addons.contract.tests.test_contract.TestContract)
2020-06-25 07:21:24,859 33 ERROR devel odoo.addons.contract.tests.test_contract: ` Editing a partner with contracts in several companies works.
2020-06-25 07:21:24,859 33 ERROR devel odoo.addons.contract.tests.test_contract: Traceback (most recent call last):
2020-06-25 07:21:24,859 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/auto/addons/contract/tests/test_contract.py", line 2513, in test_multicompany_partner_edited
2020-06-25 07:21:24,859 33 ERROR devel odoo.addons.contract.tests.test_contract: `     'parent_id': parent_partner.id,
2020-06-25 07:21:24,859 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/auto/addons/partner_autocomplete/models/res_partner.py", line 183, in write
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     res = super(ResPartner, self).write(values)
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/res_partner.py", line 570, in write
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     result = result and super(Partner, self).write(vals)
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/auto/addons/mail/models/mail_thread.py", line 321, in write
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     result = super(MailThread, self).write(values)
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/auto/addons/mail/models/mail_activity.py", line 613, in write
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     return super(MailActivityMixin, self).write(vals)
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3381, in write
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     self.recompute()
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 5308, in recompute
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     target._write(dict(vals))
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3433, in _write
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     self.check_access_rule('write')
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `   File "/opt/odoo/custom/src/odoo/odoo/models.py", line 3081, in check_access_rule
2020-06-25 07:21:24,860 33 ERROR devel odoo.addons.contract.tests.test_contract: `     + ' - ({} {}, {} {})'.format(_('Records:'), invalid.ids[:6], _('User:'), self._uid)
2020-06-25 07:21:24,861 33 ERROR devel odoo.addons.contract.tests.test_contract: ` odoo.exceptions.AccessError: ('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: Contract, Operation: write) - (Records: [101], User: 12)', None)
```

@Tecnativa TT24482

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 7.4% (20 of 270 strings)

Translation: contract-12.0/contract-12.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-12-0/contract-12-0-contract/sl/

pedrobaeza committed:
[IMP] : black, isort, prettier

pedrobaeza committed:
[MIG] contract: Migration to 13.0

- Standard procedure
- Adapt invoice creation to new account.move
- Tests adapted

pedrobaeza committed:
[IMP+REF] contract: Allow to set recurrency at header level

Big refactoring for allowing to define recurrency at header level for simplifying
the use of the module for most of the cases where you don't need different
recurrency at line level.

Commit:
[FIX] contract: multi-company assignment notification

Steps to reproduce the problem:

1. Log in as Mitchell Admin.
2. Create contract CNT-A for company CMP-A, assigned to Marc Demo.
3. Create contract CNT-B for company CMP-B, assigned to Marc Demo.
4. Run cron to create recurring invoices.

Actual results:

- Odoo sends automated assignment emails to Marc Demo, which indicate the name of the company activated for `__system__` user whlie the cron was being executed.

Expected results after this patch:

- Odoo sends automated assignment emails to Marc Demo, which indicate the invoice company.

@Tecnativa TT24657

sbejaoui committed:
[12.0][FIX] - fiscal position, payment mode and pricelist obtained according to contract company

rousseldenis committed:
[13.0][FIX] contract: Recompute recurring next date

After invoice creation, the recurring next date has to be recomputed
in order to get good values

oca-travis committed:
[UPD] Update contract.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/

CarlosRoca13 committed:
[IMP] contract: Set a condicion to avoid create lines on invoices

OCA-git-bot committed:
contract 13.0.1.0.1

victoralmau committed:
[FIX] contract: Only include 'section_and_note_one2many' js widget in view fields that contain it

OCA-git-bot committed:
contract 13.0.1.0.2

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (278 of 278 strings)

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/es_AR/

victoralmau committed:
[IMP] contract: Add followers to invoices created

oca-travis committed:
[UPD] Update contract.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
contract 13.0.1.0.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 71.6% (200 of 279 strings)

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/nl/

victoralmau committed:
[IMP] contract: Portal

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (279 of 279 strings)

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/es_AR/

pedrobaeza committed:
[IMP] contract: JS tour on test assets

oca-travis committed:
[UPD] Update contract.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
contract 13.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/

victoralmau committed:
[IMP] contract: Add contract modification

oca-travis committed:
[UPD] Update contract.pot

OCA-git-bot committed:
contract 13.0.2.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/

victoralmau committed:
[IMP] contract: Add fields to contract page in portal

oca-travis committed:
[UPD] Update contract.pot

OCA-git-bot committed:
contract 13.0.2.2.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/

victoralmau committed:
[FIX] contract: Modification mail + Auto-create follower in contract creation related to partner_id

Co-Authored-By: Pedro M. Baeza <pedro.baeza@tecnativa.com>

victoralmau committed:
[FIX] contract: Missing migration to 13.0

OCA-git-bot committed:
contract 13.0.2.2.1

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (304 of 304 strings)

Translation: contract-13.0/contract-13.0-contract
Translate-URL: https://translation.odoo-community.org/projects/contract-13-0/contract-13-0-contract/es_AR/

hparfr reviewed (approved):
LGTM

pedrobaeza commented:
We have added a lot of changes in 12.0/13.0 (some of them still ongoing), so I think this should be redone with them. I understand that may be to ask too much to include them and rework the migration, so we can take that work if you can't afford it.

ozono commented:
I'll do it. No prob.

pedrobaeza commented:
OK, thanks. I'll tell you when we finish to port all the changes to v13. They are basically:

- Portal feature
- Modification tracking

victoralmau committed:
[FIX] contract: Set groups in back_in_edit_mode link to show only correct users

OCA-git-bot committed:
contract 13.0.2.3.0

victoralmau committed:
[FIX] contract: Prevent first contract modification mail to partner

rousseldenis committed:
[13.0][FIX] contract: Impossible to create invoices from contract lines as section

OCA-git-bot committed:
contract 13.0.2.3.2

victoralmau committed:
[FIX] contract: Error in test_contract_invoice_followers

newtratip commented:
@ozono 13.0 has many changed, could you include them ?

ozono commented:
> @ozono 13.0 has many changed, could you include them ?

I'm waiting for the signal from @pedrobaeza

newtratip commented on contract/data/contract_cron.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding='UTF-8' ?>
```
```suggestion
<?xml version="1.0" encoding="UTF-8" ?>
```

newtratip commented on contract/data/contract_renew_cron.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding='UTF-8' ?>
```
```suggestion
<?xml version="1.0" encoding="UTF-8" ?>
```

newtratip commented on contract/data/mail_template.xml:
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" ?>
```
```suggestion
<?xml version="1.0" encoding="utf-8" ?>
```

newtratip commented on contract/README.rst:
```diff
@@ -0,0 +1,129 @@
+================================
```
Please generate readme follow this https://github.com/OCA/maintainer-tools/blob/master/README.md#readme-generator

newtratip commented on contract/report/report_contract.xml:
```diff
@@ -0,0 +1,167 @@
+                        <div class="row" id="invoice_info">
+                            <t t-set="total" t-value="0" />
+                            <div class="col-12">
+                                <t t-set="total" t-value="0" />
```
`<t t-set="total" t-value="0" />` is duplicated with above line 31, Can you remove one line ?

newtratip commented on contract/security/contract_security.xml:
```diff
@@ -0,0 +1,35 @@
+<?xml version='1.0' encoding='utf-8' ?>
```
```suggestion
<?xml version="1.0" encoding="utf-8" ?>
```

OCA-git-bot committed:
contract 13.0.2.3.3

newtratip commented on contract/wizards/contract_manually_create_invoice.xml:
```diff
@@ -0,0 +1,91 @@
+                                contract to invoice
+                            </span>
+                            <span
+                                attrs="{'invisible': [('contract_to_invoice_count', '&lt;', 1)]}"
```
```suggestion
                                attrs="{'invisible': [('contract_to_invoice_count', '&lt;=', 1)]}"
```

newtratip reviewed (commented):
Some review

newtratip commented:
**Functional Test**
Terminate Contract
1. Click Terminate Contract button
2. Fill Termination Date, Termination Reason, Termination Comment
3. Click Terminate Contract and click `OK` on Confirmation dialog box
4. Confirmation dialog box still show after contract terminated

My video
![Peek 2021-02-15 15-48](https://user-images.githubusercontent.com/24691983/107924375-6c2b5c00-6fa5-11eb-9d26-2f8a5cc75064.gif)

Commit:
[FIX] contract: remove groupby if it is in context

Co-authored by @Lopsanz

OCA-git-bot committed:
contract 13.0.2.3.4

Commit:
[FIX] contract: consistency between contract count and contract action

On the commercial entity, the count shows all contracts of child partners.
However clicking on the action might show no contract, as the action only
restricted the domain to the partner itself.
We extract the domain into a function to share it in both cases.

OCA-git-bot committed:
contract 13.0.2.3.5

newtratip committed:
[13.0][FIX] contract: consistency between contract count and contract action

pedrobaeza committed:
[OU-FIX] contract: Don't alter v12 recurrence mechanism

We can't guarantee the homogeneity of the existing contracts, so we
keep the line recurrence behavior.

pedrobaeza committed:
[OU-FIX] contract: Transfer contract info from invoice

OCA-git-bot committed:
contract 13.0.2.3.6

victoralmau committed:
[FIX] contract: Changed modification_mail auto to appear like email and not internal comment

victoralmau committed:
[FIX] contract: Generate recurring invoice
Remove line_ids from invoice values to prevent invoice without lines according to https://github.com/odoo/odoo/pull/67879 change

OCA-git-bot committed:
contract 13.0.2.3.7

dsolanki-initos commented:
Is it possible to get merge this PR as it is depended on other modules.

pedrobaeza commented:
@ozono all changes are merged in previous versions, so this only gets to be up to date with them and attend comments.

ozono commented:
Ok. I'll do it ASAP.

ozono committed:
[IMP] contract: black, isort, prettier

ozono committed:
[MIG] contract: Migration to 14.0

ozono commented:
All comments and suggestions have been resolved / taken into account. Thanks for the work done for V 13.0 module.

pedrobaeza commented on contract/readme/USAGE.rst:
```diff
@@ -23,9 +23,4 @@
-
-* Contracts appear in portal to following users in every contract:
-
-.. image:: ../static/src/screenshots/portal-my.png
```
Why removing these images?

ozono replied:
I think that images was meaningful when new feature was introduced. Moreover: images are not present un V. 13.0 modules (there are broken links). But, maybe, I'm wrong... If you don't agree I can take new snapshots and fix this.

pedrobaeza replied:
The idea is to showcase features in README, so they are relevant, but it's a "bug" that they weren't present in 13.0. @victoralmau can you please recover them?

ozono replied:
Ok. I'll recover them too.

ozono replied:
Done @pedrobaeza

victoralmau replied:
I have created these PR https://github.com/OCA/contract/pull/659 to add portal images in 13.0

ozono committed:
[FIX] contract: restore readme images

pedrobaeza commented:
/ocabot merge nobump
